spark/v4.0/spark/src/main/java/org/apache/iceberg/spark/source/SparkTable.java

```diff
@@ -191,10 +198,18 @@ private Schema snapshotSchema() {
     if (icebergTable instanceof BaseMetadataTable) {
       return icebergTable.schema();
     } else if (branch != null) {
-      return SnapshotUtil.schemaFor(icebergTable, branch);
+      return addLineageIfRequired(SnapshotUtil.schemaFor(icebergTable, branch));
```

Comment:
Is this needed for compacting branches?

Reply:
Correct, the procedures don't allow specifying branches right now but this is put in place so that if/when they do row lineage is covered. I preferred doing this since instead of on the PR which would add compaction for branches; it'd be easy to forget to validate that lineage is preserved as part of that.